#! /bin/sh /usr/share/dpatch/dpatch-run
## system.dpatch by Hugo Herbelin <hugo.herbelin@inria.fr>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix interpretation of ~.

@DPATCH@
diff -urNad coq-8.1~gamma~/lib/system.ml coq-8.1~gamma/lib/system.ml
--- coq-8.1~gamma~/lib/system.ml	2006-05-30 16:37:04.000000000 +0000
+++ coq-8.1~gamma/lib/system.ml	2006-11-21 21:12:08.000000000 +0000
@@ -43,7 +43,7 @@
 	  let v = safe_getenv (String.sub s (i+1) (n-i-1)) in
 	  let s = (String.sub s 0 i)^v^(String.sub s n (l-n)) in
 	  expand_macros s (i + String.length v)
-      | '~' ->
+      | '~' when i = 0 ->
 	  let n = expand_atom s (i+1) in
 	  let v =
 	    if n=i+1 then home
